neps/nep-0509.md

```diff
@@ -47,135 +48,368 @@ As a result, the team sought alternative approaches and concluded that stateless
 * As the existing logic is today, the block producer for this block waits until either all chunks are ready, or a timeout occurs, and then proposes a block containing whatever chunks are ready. Now, the notion of readiness here is expanded to also having more than 2/3 of chunk endorsements by weight.
-  * This means that if a chunk does not receive enough chunk endorsements by the timeout, it will not be included in the block. In other words, the block only contains chunks for which there is already a consensus of validity. **This is the key reason why we will no longer need challenges**.
-  * The 2/3 fraction has the denominator being the total stake assigned to validate this shard, *not* the total stake of all validators. See Chunk Validator Shuffling.
+  * This means that if a chunk does not receive enough chunk endorsements by the timeout, it will not be included in the block. In other words, the block only contains chunks for which there is already a consensus of validity. **This is the key reason why we will no longer need fraud proofs / tracking all shards**.
+  * The 2/3 fraction has the denominator being the total stake assigned to validate this shard, *not* the total stake of all validators.
 * The block producer, when producing the block, additionally includes the chunk endorsements (at least 2/3 needed for each chunk) in the block's body. The validity of the block is expanded to also having valid 2/3 chunk endorsements for each chunk included in the block.
-  * This necessitates a new block format.
   * If a block fails validation because of not having the required chunk endorsements, it is considered a block validation failure for the purpose of Doomslug consensus, just like any other block validation failure. In other words, nodes will not apply the block on top of their blockchain, and (block) validators will not endorse the block.
 
-We also propose a change to the validator roles and responsibilities. This is the list of roles after the proposal, with same and new behavior clearly labelled:
+So the high-level specification can be described as the list of changes in the validator roles and responsibilities:
 
 * Block producers:
   * (Same as today) Produce blocks, (new) including waiting for chunk endorsements
   * (Same as today) Maintain chunk parts (i.e. participates in data availability based on Reed-Solomon erasure encoding)
-  * (Same as today) Do not require tracking any shard
-  * (Same as today) Should have a higher barrier of entry for security reasons (e.g. to make block double signing harder)
+  * (New) No longer require tracking any shard
+  * (Same as today) Should have a higher barrier of entry (to keep `BlockHeader` size low and for security reasons, to make block double signing harder)
```

Comment:
how does " higher barrier of entry" helps with "to keep `BlockHeader` size low"? 🤔